dotnet/src/Extensions/Extensions.UnitTests/TemplateEngine/Prompt/Blocks/CodeBlockTests.cs

```diff
@@ -19,51 +18,40 @@ namespace SemanticKernel.Extensions.UnitTests.TemplateEngine.Prompt.Blocks;
 
 public class CodeBlockTests
 {
-    private readonly Mock<IReadOnlySKPluginCollection> _plugins;
     private readonly ILoggerFactory _logger = NullLoggerFactory.Instance;
-    private readonly Mock<IFunctionRunner> _functionRunner = new();
     private readonly Mock<IAIServiceProvider> _serviceProvider = new();
     private readonly Mock<IAIServiceSelector> _serviceSelector = new();
-
-    public CodeBlockTests()
-    {
-        this._plugins = new Mock<IReadOnlySKPluginCollection>();
-    }
+    private readonly Kernel _kernel = new(new Mock<IAIServiceProvider>().Object);
 
     [Fact]
     public async Task ItThrowsIfAFunctionDoesntExistAsync()
     {
         // Arrange
-        var functionRunner = new Mock<IFunctionRunner>();
-        var context = new SKContext(this._functionRunner.Object, this._serviceProvider.Object, this._serviceSelector.Object);
+        var context = new SKContext(this._kernel, this._serviceProvider.Object, this._serviceSelector.Object);
         var target = new CodeBlock("functionName", this._logger);
 
-        this._functionRunner.Setup(r => r.RunAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<ContextVariables>(), It.IsAny<CancellationToken>()))
-            .Returns<string, string, ContextVariables, CancellationToken>((pluginName, functionName, variables, cancellationToken) =>
-            {
-                throw new SKException("No function was found");
-            });
-
         // Act & Assert
-        await Assert.ThrowsAsync<SKException>(() => target.RenderCodeAsync(context));
+        await Assert.ThrowsAsync<KeyNotFoundException>(() => target.RenderCodeAsync(context));
     }
 
     [Fact]
     public async Task ItThrowsIfAFunctionCallThrowsAsync()
     {
         // Arrange
-        var context = new SKContext(this._functionRunner.Object, this._serviceProvider.Object, this._serviceSelector.Object, plugins: this._plugins.Object);
+        var context = new SKContext(this._kernel, this._serviceProvider.Object, this._serviceSelector.Object, plugins: this._kernel.Plugins);
+
         var function = new Mock<ISKFunction>();
+        function.Setup(x => x.Name).Returns("function");
         function
             .Setup(x => x.InvokeAsync(
                 It.IsAny<SKContext>(),
                 It.IsAny<AIRequestSettings?>(),
                 It.IsAny<CancellationToken>()))
             .Throws(new RuntimeWrappedException("error"));
```

Comment:
Presumably all of this mocking could also be replaced by an `SKFunction.FromMethod(() => throw new RuntimeWrappedException("error"))` or something similar?  Just checking; I realize the purpose of this PR isn't to clean up such tests, of which I expect there are a lot.

Reply:
Yep, sure, it can be done this way. I did not pay attention on this when refactored the tests. I will do it in the following PRs when changing unit tests.

By the way, when I tried to use the real function - `KernelFunctionFromMethod.Create((Action)(() => throw new RuntimeWrappedException("error")), "function");`, I received a `TargetInvocationException` exception that had a `RuntimeWrappedException` as an inner exception, instead of a `RuntimeWrappedException` itself. I wonder what the right behavior here would be. Should we keep throwing the `TargetInvocationException`, or should we rethrow its inner exception?

Reply:
We should use ExceptionDispatchInfo to throw the inner. On core there's an option to pass to invoke to tell it to not wrap the exception, but that's not available in netstandard2.0.